pkg/api/types.go

```diff
@@ -256,6 +258,19 @@ type SecretVolumeSource struct {
 	Target ObjectReference `json:"target"`
 }
 
+// NFS represents and NFS Mount that lasts the lifetime of a pod
+type NFSVolumeSource struct {
+	// Server is the hostname or ip address of the NFS server
+	Server string `json:"server"`
+
+	// SourcePath is the exported NFS share
+	SourcePath string `json:"sourcePath"`
+
+	// MountOptions is a comma delimited list of mount options. The available options are listed on the mount man page as arguments to -o.
+	// As an example: noatime,nodev,nosuid,ro
+	MountOptions string `json:"mountOptions"`
```

Comment:
I'll just change this to a distinct readOnly flag for this cut. And we can address the rest later.